flux-common/src/main/java/software/amazon/aws/clients/swf/flux/poller/TaskNaming.java

```diff
@@ -107,9 +107,54 @@ public static String createActivityId(String stepName, long retryAttempt, String
     }
 
     /**
-     * Given a step name, constructs a partition metadata marker name.
+     * Given a step name and information about how many metadata markers there are,
+     * constructs a partition metadata marker name.
      */
-    public static String partitionMetadataMarkerName(String stepName) {
-        return String.format("%s.%s", PARTITION_METADATA_MARKER_NAME_PREFIX, stepName);
+    public static boolean isPartitionMetadataMarker(String markerName) {
+        return markerName != null && markerName.startsWith(PARTITION_METADATA_MARKER_NAME_PREFIX);
+    }
+
+    /**
+     * Given a step name and information about how many metadata markers there are,
+     * constructs a partition metadata marker name.
+     */
+    public static String partitionMetadataMarkerName(String stepName, int subsetId, int markerCount) {
+        return String.format("%s.%s.%d.%d", PARTITION_METADATA_MARKER_NAME_PREFIX, stepName, subsetId, markerCount);
+    }
+
+    /**
+     * Given a full partition metadata marker name (as produced by TaskNaming.partitionMetadataMarker()),
```

Comment:
Not sure if we are generating javadocs here, but if you replace the references to the actual methods with something like `{@link TaskNaming::paritionMetadataMarker}` things should auto link up in the generated javadoc.

Reply:
We do generate javadocs; I'll fix it.